packages/mdc-list/foundation.ts

```diff
@@ -177,9 +178,20 @@ export class MDCListFoundation extends MDCFoundation<MDCListAdapter> {
     const isEnter = evt.key === 'Enter' || evt.keyCode === 13;
     const isSpace = evt.key === 'Space' || evt.keyCode === 32;
 
+    if (this.adapter_.isRootFocused()) {
+      if (arrowUp || isEnd) {
+        evt.preventDefault();
+        this.focusLastElement();
+      } else if (arrowDown || isHome) {
+        evt.preventDefault();
+        this.focusFirstElement();
+      }
+
+      return;
+    }
+
     let currentIndex = this.adapter_.getFocusedElementIndex();
-    let nextIndex = numbers.UNSET_INDEX;
-    if (currentIndex === numbers.UNSET_INDEX) {
+    if (currentIndex === -1) {
```

Comment:
why not just leave the -1 to `UNSET_INDEX`? Should we just get rid of `UNSET_INDEX` from constants?

Reply:
`UNSET_INDEX` is only relevant for `selectedIndex`, whereas `currentIndex` refers to the current list item where the keydown is being handled.

We still need `UNSET_INDEX` constant to reference as the list has no selection.